### DISCUSSION TITLE
Why luarocks is `dev` version?

### DISCUSSION THREAD

**Discussion by boltlessengineer** [lumen-oss/rocks.nvim#335] [Q&A] [answered]
I found `luarocks` included in `rocks.nvim` is `dev` version

```
:!which luarocks
/home/ubuntu/.local/share/nvim_rocks/rocks/bin/luarocks

:!luarocks --version
/home/ubuntu/.local/share/nvim_rocks/rocks/bin/luarocks dev
LuaRocks main command-line interface
```

This breaks `mason.nvim` because mason can’t parse the luarocks version:

```
:checkhealth mason

…
- WARNING luarocks: failed to parse version
  - ADVICE:
    - Error: ...al/share/nvim_rocks/rocks/rocks_rtp/lua/mason/health.lua:214: attempt to compare number with nil
…
```
[health.lua:214](https://github.com/williamboman/mason.nvim/blob/49ff59aded1047a773670651cfa40e76e63c6377/lua/mason/health.lua#L207)

I tried to force `rocks.nvim` to use OS installed `luarocks` with `vim.g.rocks_nvim={luarocks_binary = "/usr/bin/luarocks"}`, but I still get `dev` version of `luarocks` from `:!luarocks --version` so `mason.nvim` still fails.

Are there specific reason for using `dev` version of luarocks from `rocks.nvim`?

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

For pinning the luarocks version, I've opened https://github.com/nvim-neorocks/rocks.nvim/issues/336

> This breaks mason.nvim

I wouldn't say a warning in the health check constitutes "breaking mason.nvim".
Does anything actually break? Or is it something that could be solved by adding support for handling dev versions to mason.nvim's health checks?

**Reply by boltlessengineer** [reply]
Thank you for your quick response!

I thought this may be the reason why I couldn’t reinstall lua-language-server in my machine, but it turns out to be `mason.nvim`’s problem because I still have issues after I uninstalled all luarocks.
Sorry for stating it as “breaking” without more further research, my mistake.
